### ISSUE TITLE
Multipart parser optuout breaks admin's endpoints

### ISSUE THREAD

**Issue by EricVanDerDijs** [strapi/strapi#5177]
**Describe the bug**
If I disable multipart parsing on config\environments\development\request.json admin endpoints do not work anymore.

**Steps to reproduce the behavior**
1. Install strapi `npx create-strapi-app parser-test`
2. set this entry on config\environments\development\request.json
  ```
"parser": {
    "enabled": true,
    "multipart": false,
  }
```
3. Create a Type
4. Try to create a new object of the newly created type, and receive this error message: 
> When using multipart/form-data you need to provide your data in a JSON 'data' field.

**Expected behavior**
Data should be saved. please read **Additional context**.

**System**
- Node.js version: 12.13.1
- NPM version: 6.12.1
- Strapi version: 3.0.0-beta.18.7
- Database: sqlite (quickstart), but also happens on mongo
- Operating system: windows 10

**Additional context**
**IMPORTANT! :** I understand that turning off multipart parsing should affect admin since Types can have files in them, but my actual problem is that I need to do some processing on a file sent by the client via streaming, so, if I could:

a. Set just one endpoint to opt out from the multipart parsing (the only endpoint where I actually need that), **or**.
b. Keep parser working for the admin endpoints, even if multipart parsing is disabled.

Then my issue would be solved.

Really nice plattaform by the way!

**Comment by lauriejim** [comment]
Hello @EricVanDerDijs thank you for reporting that.
You want to process your image for one of your REST API routes or for the admin panel?
And can you precise a little bit what you are trying to do (steps you would like to achieve)

**Comment by EricVanDerDijs** [comment]
Hi @lauriejim, of course, let me explain what I did, maybe that could clarify this use case:

**TL;DR:** I intend to process a file on one custom REST API route.

1. I added an endpoint to an entity's router, i.e.: POST - /entity/text-processing, it receives a file and process it, doing batch DB update.
2. Since strapi automatically parses the content of the multipart request, I had to disable multipart parsing so I could do the processing of the data via streaming. **If I could optout of the multipart parsing only this endpoint that would be awasome.**
3. Then after disabling strapi's multipart parsing the admin stopped working.

**Comment by lauriejim** [comment]
Okay I think you can try to use this part of the documentation - https://strapi.io/documentation/3.0.0-beta.x/concepts/middlewares.html#custom-middlewares-structure
This should override the source one.

Here is the source code of the parser middleware - https://github.com/strapi/strapi/tree/master/packages/strapi/lib/middlewares/parser
You can see a condition to no use it for GraphQL endpoint, you can do the same for yours.

**Comment by EricVanDerDijs** [comment]
Hello @lauriejim , I tried your approach but for some reason I have not been able to overide the parser middleware. my code:

config\environments\development\middleware.json
```javascript
{
  "parser": {
    "enabled": true
  }
}
```
middlewares\parser\index.js
```javascript
module.exports = strapi => {
  return {
    /**
     * Initialize the hook
     */

    initialize() {
      strapi.app.use((ctx, next) => {
      });
    },
  };
};
```

if I change **parser** for **test** the middleware is hit, but for parser it does not work.

Mayby this has something to do with it?

config\environments\development\request.json
```json
{
  "...otherConfigs": "...", 
  "parser": {
    "enabled": true,
    "multipart": true,
    "formidable": {
      "maxFileSize": 524288000
    }
  }
}
```

**Comment by lauriejim** [comment]
Arf after digging into the code, we updated the load of middlewares - https://github.com/strapi/strapi/blob/master/packages/strapi/lib/core/load-middlewares.js#L20
Due to the `Promise.all` the middleware that is used could be random...
I have to discuss with the team about the management we want about that.

So I suggest you disable the `parser` middleware and create a `customParser` middleware that is a copy of the original one.

Sorry for the non-correct answer.

**Comment by EricVanDerDijs** [comment]
That didn't work either, after setting **config\environments\development\middleware.json** to
```json
{
  "parser": {
    "enabled": false
  },
  "customParser": {
    "enabled": true
  }
}
```

the default parser is still beign used, so, I'm unable to disable it or override it.

Reading the file that you refered to, it seems that the middlewares are not disabled in that file. I guess that if instead of Promise.all they were loaded one after the other, according to an importance rule, then It would fix the overiding issue, right? or maybe I'm not seeing something.

**Comment by EricVanDerDijs** [comment]
@lauriejim sorry to bother you, but, do you think that there might be a solution to this?

**Comment by lauriejim** [comment]
Hello! I just tried and it works well.

Here is what I did -  https://www.loom.com/share/81cbb4638fd5455eb7ab9af69ebc37c0

After that in the custom middleware, add an exception for the endpoint you want, like `/graphlq`

**Comment by EricVanDerDijs** [comment]
Ohhhhhh ok, thanks a lot! that did work!

I was trying to disable the parser middleware through the middleware env config following the docs, but doing it your way did the trick!!
